DependencyInjection/JMSSerializerExtension.php

```diff
@@ -143,6 +144,20 @@ private function loadInternal(array $config, ScopedContainer $container, array $
             $container
                 ->getDefinition('jms_serializer.accessor_strategy.default')
                 ->replaceArgument(0, new Reference($config['expression_evaluator']['id']));
+
+            if (is_a($container->getDefinition($config['expression_evaluator']['id'])->getClass(), CompilableExpressionEvaluatorInterface::class, true)) {
```

Comment:
just replace this with `findDefinition`

Reply:
It's going to have the same problem.  The `ScopedContainer::findDefinition()` method also pushes the ID through `DIUtils::getRealId()`.

Reply:
i have tried locally and it works. the issue is that `getDefinition` tries to find a service that has been aliased... i know `ScopedContainer` is a monster mess, but "somehow it works"